code/01_Functions_01_schedule_workbench_job.R

```diff
@@ -0,0 +1,217 @@
+################################################################################
+# Name of file:       schedule_workbench_job.R
+# Type of script:     R
+#
+# Original author:    Terry McLaughlin
+#
+# Written/run on: R version 4.1.2 (2021-11-01) -- "Bird Hippie"
+# Platform: x86_64-pc-linux-gnu (64-bit)
+#
+# Function to programmatically schedule the launch of workbench jobs on the
+# Kubernetes cluster.
+################################################################################
+
+### 00 schedule_workbench_job()  ----
+
+# A function to programmatically schedule the launch of a workbench job on the
+# Kubernetes cluster
+schedule_workbench_job <- function(job_name,       # Name to give the Workbench Job
+                                   schedule_name,  # The name of the event loop to schedule the Workbench Job on        
+                                   due,            # Date / time (as POSIXct) to run the Workbench Job
+                                   rpt = NULL,     # Repeat every n seconds thereafter
+                                   project_path,   # Path to project / working directory
+                                   script,         # Relative path to R script to execute
+                                   n_cpu = NULL,   # Number of CPUs to request
+                                   n_ram = NULL) { # Amount of memory (MB) to request
```

Comment:
I can see why you've done it like this but would it make more sense to provide broad consistency in terms of where arguments are placed by focussing on the current task first, e.g. giving the schedule a name before referencing the job_name?

```suggestion
schedule_workbench_job <- function(
                                   schedule_name,  # The name of the event loop to schedule the Workbench Job on        
                                   due,            # Date / time (as POSIXct) to run the Workbench Job
                                   job_name,       # Name to give the Workbench Job
                                   rpt = NULL,     # Repeat every n seconds thereafter
                                   project_path,   # Path to project / working directory
                                   script,         # Relative path to R script to execute
                                   n_cpu = NULL,   # Number of CPUs to request
                                   n_ram = NULL) { # Amount of memory (MB) to request
```

Reply:
Sure, but I'd suggest going one step further, and keep all the _schedule_ arguments together:

```suggestion
schedule_workbench_job <- function(
                                   schedule_name,  # The name of the event loop to schedule the Workbench Job on        
                                   due,            # Date / time (as POSIXct) to run the Workbench Job
                                   rpt = NULL,     # Repeat every n seconds thereafter
                                   job_name,       # Name to give the Workbench Job
                                   project_path,   # Path to project / working directory
                                   script,         # Relative path to R script to execute
                                   n_cpu = NULL,   # Number of CPUs to request
                                   n_ram = NULL) { # Amount of memory (MB) to request
```